example/v1/types_stable.go

```diff
@@ -63,6 +67,40 @@ const (
 	TechPreviewOnlyValue EvolvingDiscriminator = "TechPreviewOnlyValue"
 )
 
+// CELUnion demonstrates how to use a discriminated union and how to validate it using CEL.
```

Comment:
if the RequiredMember is set, you want to verify the self.type.

Reply:
First part of the ternary condition, `has(self.type)` fails if type isn't set, at which point we fall into `!has(self.requiredMember)` which fails if `RequiredMember` is set, which then fails the validation and returns the error.

Reply:
We have `should not allow a member without the optional type value` as a test that checks the optional but I can add for required too

Reply:
Added explicit tests for both now